### ISSUE TITLE
Design: RAG data entry mockup

### ISSUE THREAD

**Issue by mairin** [projectatomic/studio-extension#67]
UI for users to connect to vector db of their data for RAG usage of model

**Comment by mairin** [comment]
Notes for 6 Feb program call:

Concept of an embedding model is unique to RAG patterns, may have implications on AI Studio UI.

RAG pattern:
- choose gen AI model AND an embedding model

Erik has a deck for vector dbs, can share:

[https://docs.google.com/presentation/d/1FkprZvdfkYiBjmWW5xDgOPc29XymVIIMzuzU0ipaA-A/edit#slide=id.p](https://www.google.com/url?q=https://docs.google.com/presentation/d/1FkprZvdfkYiBjmWW5xDgOPc29XymVIIMzuzU0ipaA-A/edit%23slide%3Did.p&sa=D&source=docs&ust=1707232919985485&usg=AOvVaw18QRtG0Bn9cnUI6EyEZiNm)

Embedding model is the indexing thing... ask question in chat with model. For RAG - that question gets run through this embedding model, puts it into the embedding space.... when you index all your docs, you run them through that same model. It'sl ike a database, the index part is embedding vectors. Instead of doing exact lookup on a key, it's looking for documents with vectors closest to the thing.

**Comment by lstocchi** [comment]
@mairin is there any draft mockup i can start looking at? I was updating the catalog but i need to work on this before adding the rag recipe

**Comment by mairin** [comment]
Hi Luca, nothing yet :-( I spoke to Jeff about this yesterday. After I get
the extensions mocks done today its next on my list. ~m

**Comment by vrothberg** [comment]
@nichjones1 @mairin @lstocchi shall we reassign the issue?  Do we have mock ups?

**Comment by nichjones1** [comment]
@vrothberg I've added Emma as an assignee.  We will discuss at our next checkin

**Comment by vrothberg** [comment]
I do not know, but @mairin @nichjones1 or @slemeur may know.

**Comment by jeffmaury** [comment]
If this board is about RAG applications, the mock ups are about image generation which is a different topic.

**Comment by ekidneyrh** [comment]
Ok. What is the user flow to get to this page? I'm not familiar with RAG applications and the layout, so I'll have to do some research beforehand.

**Comment by ekidneyrh** [comment]
Ok so I had another look at this. I suppose my question would be what fields / information do we need from the user? Is this something that will be on a form page - and if so where within the application does it fit in? Is it going to be in Catalog > Models > Import? Or is it going to have its own menu item to navigate to it? I need more information about this before I can create a mockup.

**Comment by jeffmaury** [comment]
I woujld add that RAG recipes would need:
- an additional embedding model to start from

We may also add a datastore from which the embeddings will be computed but I'm not sure this may not be handled by the recipe itself

**Comment by MariaLeonova** [comment]
@slemeur @jeffmaury can you please confirm that the earlier comments in this ticket are still valid and elaborate some more on the exact requirements? Who is the user and what are they trying to do through RAG data entry?

**Comment by jeffmaury** [comment]
> @slemeur @jeffmaury can you please confirm that the earlier comments in this ticket are still valid and elaborate some more on the exact requirements? Who is the user and what are they trying to do through RAG data entry?

From my POV yes but need @slemeur confirmation

**Comment by MariaLeonova** [comment]
is this RAG demo relevant here? https://youtu.be/mouu4PR5Dqg?si=eLBieGqH0-ZRgP4L

**Comment by MariaLeonova** [comment]
@jeffmaury can you please help me understand, who is is the user, who wants to connect their vector db and what are they doing? It could be in a from of a user story: As a ..., I want to ..., so that I can ....
I.e. is this an additional story to the fine-tuning or is this part of the recipe or is it something else entirely? 

It would be helpful to understand, what screen are they starting from and what their goals are when they are bringing the database in.

**Comment by jeffmaury** [comment]
> @jeffmaury can you please help me understand, who is is the user, who wants to connect their vector db and what are they doing? It could be in a from of a user story: As a ..., I want to ..., so that I can .... I.e. is this an additional story to the fine-tuning or is this part of the recipe or is it something else entirely?
> 
> It would be helpful to understand, what screen are they starting from and what their goals are when they are bringing the database in.

Yes that's not entirely clear to me but I can say that it's disconnected from the fine tuning workflow
For my POV, is is either:

- part of the workflow starting RAG based recipes
- a new workflow to help users configure their RAG based AI application

Maybe @slemeur can help

HTH
